File: backend/src/main/java/com/stockflow/backend/entity/Product.java
package com.stockflow.backend.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.math.BigDecimal;

@Entity
@Table(name = "product")
@AllArgsConstructor
@NoArgsConstructor
@Getter @Setter
public class Product {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String name;

    @Column(nullable = false, unique = true)
    private String code;

    @Column(nullable = false)
    private String description;

    @Column(nullable = false, precision = 10, scale = 2)
    private BigDecimal price;

    @Column(name = "offer_price", precision = 10, scale = 2)
    private BigDecimal offerPrice;

    @Column(name = "stock", nullable = false)
    private int stock;

    @Column(name = "stock_min", nullable = false)
    private int stockMin;

    @Column(nullable = false)
    private boolean active = true;

    @ManyToOne
    @JoinColumn(name = "category_id", nullable = false)
    private Category category;

    @ManyToOne
    @JoinColumn(name = "supplier_id", nullable = false)
    private Supplier supplier;
}

File: frontend/src/types/product.ts
import type { CategoryResponse } from './category';
import type { SupplierResponse } from './supplier';

export interface ProductResponse {
  id: number;
  name: string;
  code: string;
  description: string;
  price: number;
  offerPrice: number | null;
  stock: number;
  minStock: number;
  active: boolean;
  category: CategoryResponse;
  supplier: SupplierResponse;
}

export interface ProductCreateRequest {
  name: string;
  code: string;
  description: string;
  price: number;
  offerPrice: number | null;
  initialStock: number;
  minStock: number;
  categoryId: number;
  supplierId: number;
}